google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineAction.java

```diff
@@ -91,14 +94,30 @@ protected void executeProcess(
 
     consoleLogLn("Executing: " + commandLine.getCommandLineString());
 
-    final Process process = commandLine.createProcess();
-    final ProcessHandler processHandler = new OSProcessHandler(process,
-        commandLine.getCommandLineString());
+    Process process = commandLine.createProcess();
+    processHandler = new OSProcessHandler(process, commandLine.getCommandLineString());
     loggingHandler.attachToProcess(processHandler);
     processHandler.addProcessListener(listener);
+
+    // mark process as completed by setting handler to null when terminated
+    processHandler.addProcessListener(new ProcessAdapter() {
+      @Override
+      public void processTerminated(ProcessEvent event) {
+        processHandler = null;
+      }
+    });
+
     processHandler.startNotify();
   }
 
+  protected void cancel() {
```

Comment:
I was just trying to be consistent executeProcess in the same class
`protected void executeProcess() {...}`